src/arithmetic/arithmetic_expression.c

```diff
@@ -715,7 +396,43 @@ void AR_EXP_Free(AR_ExpNode *root) {
 	rm_free(root);
 }
 
-/* Mathematical functions - numeric */
+bool AR_FuncExists(const char *func_name) {
+	char lower_func_name[32] = {0};
+	size_t lower_func_name_len = 32;
+	_toLower(func_name, &lower_func_name[0], &lower_func_name_len);
+	void *f = TrieMap_Find(__aeRegisteredFuncs, lower_func_name, lower_func_name_len);
+	return (f != TRIEMAP_NOTFOUND);
+}
+
+void AR_RegisterFuncs() {
+	struct RegFunc {
+		const char *func_name;
+		AR_Func func_ptr;
+	};
+
+	struct RegFunc functions[39] = {
+		{"add", AR_ADD}, {"sub", AR_SUB}, {"mul", AR_MUL}, {"div", AR_DIV}, {"abs", AR_ABS}, {"ceil", AR_CEIL},
+		{"floor", AR_FLOOR}, {"rand", AR_RAND}, {"round", AR_ROUND}, {"sign", AR_SIGN}, {"left", AR_LEFT},
+		{"reverse", AR_REVERSE}, {"right", AR_RIGHT}, {"ltrim", AR_LTRIM}, {"rtrim", AR_RTRIM}, {"substring", AR_SUBSTRING},
+		{"tolower", AR_TOLOWER}, {"toupper", AR_TOUPPER}, {"tostring", AR_TOSTRING}, {"trim", AR_TRIM}, {"contains", AR_CONTAINS},
+		{"starts with", AR_STARTSWITH}, {"ends with", AR_ENDSWITH}, {"id", AR_ID}, {"labels", AR_LABELS}, {"type", AR_TYPE}, {"exists", AR_EXISTS},
+		{"timestamp", AR_TIMESTAMP}, {"and", AR_AND}, {"or", AR_OR}, {"xor", AR_XOR}, {"not", AR_NOT}, {"gt", AR_GT}, {"ge", AR_GE},
+		{"lt", AR_LT}, {"le", AR_LE}, {"eq", AR_EQ}, {"neq", AR_NE}, {"case", AR_CASEWHEN}
+	};
+
+	char lower_func_name[32] = {0};
+	size_t lower_func_name_len = 32;
+
+	for(int i = 0; i < 39; i++) {
+		_toLower(functions[i].func_name, &lower_func_name[0], &lower_func_name_len);
+		_AR_RegFunc(lower_func_name, lower_func_name_len, functions[i].func_ptr);
+		lower_func_name_len = 32;
+	}
+}
+
+//==============================================================================
+//=== Mathematical functions - numeric =========================================
+//==============================================================================
```

Comment:
I personally think it would be cleaner to move all the operation funcions to a `funcs.c` file (matching their header) and just have AR_ExpNode logic and such here.

Reply:
yeah, I thought about it, didn't want to include that change in this PR,
I think there should be even greater separation as the comment implies.